packages/anoncreds/src/formats/LegacyIndyCredentialFormatService.ts

```diff
@@ -0,0 +1,603 @@
+      AnonCredsCredentialRequestMetadataKey
+    )
+
+    const registryService = agentContext.dependencyManager.resolve(AnonCredsRegistryService)
+    const anonCredsHolderService =
+      agentContext.dependencyManager.resolve<AnonCredsHolderService>(AnonCredsHolderServiceSymbol)
+
+    if (!credentialRequestMetadata) {
+      throw new CredentialProblemReportError(
+        `Missing required request metadata for credential with id ${credentialRecord.id}`,
+        { problemCode: CredentialProblemReportReason.IssuanceAbandoned }
+      )
+    }
+
+    const anonCredsCredential = attachment.getDataAsJson<AnonCredsCredential>()
+
+    // We can use the same registry for the credential definition and revocation registry as they MUST have the same issuerId
+    const registry = registryService.getRegistryForIdentifier(agentContext, anonCredsCredential.cred_def_id)
+
+    const credentialDefinitionResult = await registry.getCredentialDefinition(
+      agentContext,
+      anonCredsCredential.cred_def_id
+    )
+    if (!credentialDefinitionResult.credentialDefinition) {
+      throw new AriesFrameworkError(
+        `Unable to resolve credential definition ${anonCredsCredential.cred_def_id}: ${credentialDefinitionResult.resolutionMetadata.error} ${credentialDefinitionResult.resolutionMetadata.message}`
+      )
+    }
+
+    const revocationRegistryResult = anonCredsCredential.rev_reg_id
+      ? await registry.getRevocationRegistryDefinition(agentContext, anonCredsCredential.rev_reg_id)
+      : null
+    if (revocationRegistryResult && !revocationRegistryResult.revocationRegistryDefinition) {
+      throw new AriesFrameworkError(
+        `Unable to resolve revocation registry definition ${anonCredsCredential.cred_def_id}: ${credentialDefinitionResult.resolutionMetadata.error} ${credentialDefinitionResult.resolutionMetadata.message}`
+      )
+    }
+
+    if (!credentialRecord.credentialAttributes) {
+      throw new AriesFrameworkError(
+        'Missing credential attributes on credential record. Unable to check credential attributes'
+      )
+    }
+
+    // assert the credential values match the offer values
+    const recordCredentialValues = convertAttributesToCredentialValues(credentialRecord.credentialAttributes)
+    assertCredentialValuesMatch(anonCredsCredential.values, recordCredentialValues)
+
+    const credentialId = await anonCredsHolderService.storeCredential(agentContext, {
+      credentialId: utils.uuid(),
+      credentialRequestMetadata,
+      credential: anonCredsCredential,
+      credentialDefinitionId: credentialDefinitionResult.credentialDefinitionId,
+      credentialDefinition: credentialDefinitionResult.credentialDefinition,
+      revocationRegistry: revocationRegistryResult?.revocationRegistryDefinition
+        ? {
+            definition: revocationRegistryResult.revocationRegistryDefinition,
+            id: revocationRegistryResult.revocationRegistryDefinitionId,
+          }
+        : undefined,
```

Comment:
```suggestion
      revocationRegistryDefinition: revocationRegistryResult?.revocationRegistryDefinition,
```

Something like that. It might make the object a bit too verbose in the metadata though.

Reply:
Where do we put the id in that case? On the same level?

Reply:
Ah yeah sorry. I mistyped the suggestion. I wanted to just do the `Result` type. But I am fine with you solution. Please disregard :).